include/rmm/device_scalar.hpp

```diff
@@ -92,6 +146,37 @@ class device_scalar {
 
  private:
   rmm::device_buffer buff{sizeof(T)};
+
+  template<bool synchronize>
+  inline T _value(cudaStream_t stream) const {
+    T host_value{};
+    _memcpy<synchronize>(&host_value, buff.data(), sizeof(T), stream);
+    return host_value;
+  }
+
+  template<bool synchronize>
+  inline void _set_value(T host_value, cudaStream_t stream) {
+    _memcpy<synchronize>(buff.data(), &host_value, sizeof(T), stream);
+  }
+
+  template<bool synchronize>
+  inline void _memcpy(void *dst, const void *src, size_t count, cudaStream_t stream) const {
+    auto status = cudaMemcpyAsync(dst, src, count, cudaMemcpyDefault, stream);
```

Comment:
@harrism what's the guidance for cuda driver calls? should these use the [`RMM_CHECK_CUDA` macro](https://github.com/rapidsai/rmm/blob/branch-0.11/include/rmm/rmm.hpp#L43)?

Reply:
Yes. That way we can update the macro to throw exceptions in a future PR.

Reply:
For the sake of consistency and avoiding scope creep, I'd like to defer that change to a future PR. `device_buffer` uses similar error handles as to what is done here, and this class sub-classes `device_buffer`. To change only this function would not be ideal, and to update `device_buffer` would be out of scope. I know @jrhemstad has plans to overhaul exception handling in RMM. Technically I'm not changing error handling in this class, I'm only moving the code and re-using it in a very similar method. Thoughts?

Reply:
All we're asking is that you use RMM_CHECK_CUDA. That's not scope creep.

Reply:
`RMM_CHECK_CUDA` returns an error code, which won't work here.